### ISSUE TITLE
Import single stylus file with npm

### ISSUE THREAD

**Issue by mesqueeb** [stylus/stylus#2402]
Dear Stylus community.
I have a bunch of stylus files "helpers" that I always manually copy paste in projects.
Is there a way to have an `npm module` with just a stylus file so I can import it into a project and keep a version history on Github?

Any advice much appreciated!

**Comment by Mottie** [comment]
I was also looking for a solution.

The [try Stylus site](http://stylus-lang.com/try.html) uses a [minified version of stylus](http://stylus-lang.com/try/stylus.min.js) for the online demo. I was not able to find the source, nor was I able to duplicate the minified file using browserify or rollup. I ended up using that file to make a new repository and published it to npm - https://www.npmjs.com/package/stylus-lang-bundle.

**Comment by mesqueeb** [comment]
We really need an `spm` the stylus package manager. or `cpm` the css package manager`....

**Comment by mesqueeb** [comment]
And have an npm package with just stylus files?
It's a tad annoying to setup webpack etc. to know how to resolve paths to the correct node module inside stylus files I'd argue. (whereas in js files you can just `import something from 'a-package'`)

Do you have any advice on how to keep these "paths" to write inside the stylus file as standalone as possible? Eg. knowing the correct node module path irrelevant of its location inside a project?

**Comment by pakastin** [comment]
No, just import straight from GitHub – you can also pin version with `npm i pakastin/module#v0.1.2`

**Comment by mesqueeb** [comment]
I mean, i'm confused if you import straight from github, what to add in your `app.styl` file. Could you give an example?

**Comment by pakastin** [comment]
Well, with `npm i pakastin/module#v0.1.2` just use `@import 'module/path-to-file/file.styl`

**Comment by pakastin** [comment]
hmm.. not sure, but you might probably need to also do `stylus -u module`, just like for example with `nib` as well..

**Comment by mesqueeb** [comment]
Thanks! This is exactly what I meant. If you have to hard-code each `path-to-file` you'll get into a lot of trouble when moving around some files later. Where `node_modules` is just magically resolved.

So my question remains:

> It's a tad annoying to setup webpack etc. to know how to resolve paths
> Do you have any advice on how to keep these "paths" to write inside the stylus file as standalone as possible? Eg. knowing the correct node module path irrelevant of its location inside a project?

**Comment by pakastin** [comment]
Sorry, I don't understand the question..?

**Comment by mesqueeb** [comment]
With a js package:
- cli: `npm i my-package`
- js file: `import myPackage from 'my-package'`

With stylus files you say:
- cli: `npm i pakastin/module`
- stylus file: `@import 'module/path-to-file/file.styl`

What I mean is that anywhere in the project I can `import myPackage from 'my-package'` but this is not the case for stylus files... how does the stylus file know how to point to the correct `module` folder regardless of its location inside the project.

**Comment by atilkan** [comment]
@pakastin He is asking how to import from node_modules?

**Comment by mesqueeb** [comment]
@atillan exactly! I know how to work with js files and npm, but no idea of how stylus would resolve a path to go automatically look in the node_modules folder.

Do we have to hard code a path relative from the app.styl file to the correct styl file installed via npm? In this case my paths would look something like: `@import '../../node_modules/my-stylus-module/my-file.styl`

Not very elegant as you can see.......

Is this the way people manage their personal libraries of stylus helper functions and files best?
1. Upload on npm and 2. Use a path like that to the file inside node_modules

I just feel like there gotta be a better way and i'm sure a million developers are facing this problem. How is everyone doing it? ;D

**Comment by atilkan** [comment]
You can fiddle around with nib. 
https://github.com/stylus/nib

**Comment by mesqueeb** [comment]
@atilkan I see that `nib` offers some helpers and mixins, but I was looking for a package manager where I can easily import my own helpers, mixins and styl files. Any advice on that?

@pakastin Was I able to clarify myself? : D

**Comment by pakastin** [comment]
you can define paths relative to a node module:
```
module/css/index.styl
```
–>
```
@import 'module/css/index.styl';
```

**Comment by mesqueeb** [comment]
@pakastin Will stylus files automatically resolve paths to go look in the `node_modules` folder? Or is this something I would need to set up in webpack or something?

**Comment by mesqueeb** [comment]
@pakastin So it's just .... magic? 🧙🏻‍♂️

**Comment by pakastin** [comment]
Hmm.. you might need to use `stylus -u modulename` actually

**Comment by TalD** [comment]
Example usage:
let's say you have a task in your `package.json` called `build-css`
You might run this task by typing `$ npm run build-css`
your `package.json` would (roughly speaking) look like this:

`{ "scripts": { "build-css": "stylus -c src/path/to/my/stylus/main.styl -o src/path/to/my/output/main.css -I node_modules/some/path/to/your/resource/" } }`

This line tells your `Stylus` to compile `main.styl` to `main.css` and also include a `lookup path`
According to the docs:

`-I, --include <path>    Add <path> to lookup paths`

Now let's say in the base directory of that node package you have a `reset.styl`

In your `main.styl` file you can simply `@import "reset"` and because of the `lookup path`, the compiler will know to import / compile that bit of code.

Worth noting. If you want to import `CSS` (as opposed to stylus) from your node modules (or anywhere for that matter)

You need the following flag: `--include-css`

Your `package.json` script would now look like this:

`{ "scripts": { "build-css": "stylus -c src/path/to/my/stylus/main.styl -o src/path/to/my/output/main.css -I node_modules/some/path/to/your/resource/ --include-css" } }`

**Comment by daGaiGuanYu** [comment]
It seems like a webpack problem. Because you can compile ```xyz.styl``` by ```npx stylus xyz.styl```. And ```xyz.styl``` looks like:
``` styl
@import 'your-helper' // your helper is in node_modules
```
I'm using snowpack, and snowpack act simalarly to webpack.
